test/Sync.t.sol

```diff
@@ -0,0 +1,244 @@
+
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true});
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency0);
+
+        swapRouter.swap(key, params, testSettings, new bytes(0));
+        (uint256 balanceCurrency0) = currency0.balanceOf(address(manager));
+        assertEq(manager.getReserves(currency0), balanceCurrency0); // Reserves are up to date since settle was called.
+    }
+
+    function test_settle_withNoStartingBalance() public {
+        assertEq(currency2.balanceOf(address(manager)), uint256(0));
+
+        (Currency cur0, Currency cur1) = currency0 < currency2 ? (currency0, currency2) : (currency2, currency0);
+        PoolKey memory key2 =
+            PoolKey({currency0: cur0, currency1: cur1, fee: 3000, tickSpacing: 60, hooks: IHooks(address(0))});
+
+        manager.initialize(key2, SQRT_RATIO_1_1, new bytes(0));
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency2);
+        modifyLiquidityRouter.modifyLiquidity(key2, IPoolManager.ModifyLiquidityParams(-60, 60, 100), new bytes(0));
+        (uint256 balanceCurrency2) = currency2.balanceOf(address(manager));
+        assertEq(manager.getReserves(currency2), balanceCurrency2);
+    }
+
+    function test_settle_revertsIfSyncNotCalled() public {
+        Actions[] memory actions = new Actions[](1);
+        bytes[] memory params = new bytes[](1);
+
+        actions[0] = Actions.SETTLE;
+        params[0] = abi.encode(currency0);
+
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        router.executeActions(actions, params);
+    }
+
+    /// @notice When there is no balance and reserves are set to type(uint256).max, no delta should be applied.
+    function test_settle_noBalanceInPool_shouldNotApplyDelta() public {
+        assertEq(currency2.balanceOf(address(manager)), uint256(0));
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency2);
+
+        manager.sync(currency2);
+        assertEq(manager.getReserves(currency2), 0);
+
+        Actions[] memory actions = new Actions[](2);
+        bytes[] memory params = new bytes[](2);
+
+        actions[0] = Actions.SETTLE;
+        params[0] = abi.encode(currency2);
+
+        actions[1] = Actions.ASSERT_DELTA_EQUALS;
+        params[1] = abi.encode(currency2, address(router), 0);
+
+        router.executeActions(actions, params);
+    }
+
+    /// @notice When there is no actual balance in the pool, but reservesBefore is outdated (as sync has not been called),
+    /// a delta of type(uint256).max - reservesBefore should not be applied, and no amount should be takeable from the pool.
```

Comment:
```suggestion
    /// a delta of ZERO_BALANCE - reservesBefore should not be applied when calling settle, and no amount should be takeable from the pool.
```

Reply:
or something like this... took me a long time to understand what this test is actually testing

Reply:
ok I updated extensively haha